cachi2/core/models/input.py

```diff
@@ -105,6 +105,12 @@ def check_package_paths(cls, package: PackageInput, values: dict) -> PackageInpu
                 )
         return package
 
+    @pydantic.validator("packages")
+    def non_empty_packages(cls, packages: list[PackageInput]):
+        if len(packages) == 0:
+            raise ValueError("Packages cannot be empty")
+        return packages
```

Comment:
Oof, this will break a lot of tests :smile: 

Maybe this should be validated on the input-parsing side? I don't necessarily think that having 0 packages makes the Request invalid

Reply:
In fact, do we want to validate it at all? Maybe there will be a use case for a package-less cachi2.args file e.g. to make HACBS use network isolation even when no pre-fetch is needed?

Reply:
My reasoning here was: if the user has an invalid args file, we want to show it to him right away. And probably a good definition of a valid file is one that has minimally one package in it.  If we allow it to be empty, the user could create a valid request by defining the parameters via stdin while having an empty file. 

My main concern here is to leave less room for confusion.  I think it's fine to validade it in the CLI level, though.

+1 for having an extra parameter in case we need packageless requests.

Reply:
So if I'm understanding it correctly, you suggest the validation should fail when the file is empty, even when the user does specify a package via the CLI args?

In that case we would need this validator in ArgsFile, not in Request